platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/ProfileInstallCommand.java

```diff
@@ -0,0 +1,231 @@
+
+import ddf.security.Subject;
+import ddf.security.SubjectUtils;
+
+import static org.fusesource.jansi.Ansi.ansi;
+
+@Command(scope = "admin", name = "profile-install", description = "Installs the apps and features from a given install profile")
+@Service
+public class ProfileInstallCommand extends AbstractProfileCommand {
+    @Argument(index = 0, name = "profileName", description = "Name of install profile to use", required = true, multiValued = false)
+    String profileName;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProfileInstallCommand.class);
+
+    @Inject
+    protected ConfigurationAdmin configAdmin;
+
+    private FeaturesService featuresService;
+
+    public ProfileInstallCommand() {
+        featuresService = (FeaturesService) getService(FeaturesService.class);
+
+        if (featuresService == null) {
+            LOGGER.debug("Features Service was null");
+        }
+    }
+
+    @Override
+    protected void doExecute(ApplicationService applicationService)
+            throws ApplicationServiceException {
+
+        if (!profileExists(applicationService.getInstallationProfiles(), profileName) && showDevProfiles) {
+            Map<String, List<String>> devProfile = getDevProfile(profileName);
+
+            startApps(applicationService, devProfile.get(DEV_PROFILE_START_APPS));
+
+            if (featuresService != null) {
+                uninstallInstallerModule();
+                installFeatures(devProfile.get(DEV_PROFILE_INSTALL_FEATURES));
+                uninstallFeatures(devProfile.get(DEV_PROFILE_UNINSTALL_FEATURES));
+            }
+
+            stopBundles(devProfile.get(DEV_PROFILE_STOP_BUNDLES));
+
+            console.println(ansi().fgBrightGreen().a("Installation complete.").reset());
+        } else {
+            Feature profile = getProfile(applicationService, profileName);
+            List<String> profileApps = profile.getDependencies().stream()
+                    .map(Dependency::getName)
+                    .collect(Collectors.toList());
+
+            startApps(applicationService, profileApps);
+            uninstallInstallerModule();
+        }
+    }
+
+    private void startApps(ApplicationService applicationService, List<String> startupApps)
+            throws ApplicationServiceException {
+        if (startupApps != null) {
+            printSectionHeading("Starting Applications");
+            startupApps.forEach(application -> {
+                printItemStatusPending("Starting: ", application);
+                executeAsSystem(() -> {
+                    applicationService.startApplication(application);
+                    return true;
+                });
+                if (applicationService.isApplicationStarted(applicationService.getApplication(application))) {
+                    printItemStatusSuccess("Started: ", application);
+                } else {
+                    printItemStatusFailure("Start Failed: ", application);
+                }
```

Comment:
Should we return a failure notification out such that the above method not continue with the next section to process?